packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -18,6 +19,42 @@ export class Permissions {
     this.superAdmin = superAdmin;
   }
 
+  public canViewCreateDataSourceModal = (project?: string): boolean => {
+    return this.checkProjectFilterPermission(
+      {
+        projects: project ? [project] : [],
+      },
+      "createDatasources"
+    );
+  };
+
+  public canCreateDataSource = (
+    datasource: Pick<DataSourceInterface, "projects">
+  ): boolean => {
+    return this.checkProjectFilterPermission(datasource, "createDatasources");
+  };
+
+  public canUpdateDataSourceParams = (
+    datasource: Pick<DataSourceInterface, "projects">
+  ): boolean => {
+    return this.checkProjectFilterPermission(datasource, "createDatasources");
+  };
```

Comment:
This deviates slightly from the other `canUpdateXYZ` methods in that it doesn't take in the `existing` and `updates`. The logic within the `putDatasource` method is pretty deep, so it feels safer to leave it as is and go this route, but open to opinions.